client/ayon_flame/plugins/publish/collect_shots.py

```diff
@@ -0,0 +1,342 @@
+    """Collect new shots."""
+
+    order = order = pyblish.api.CollectorOrder - 0.095
+    label = "Collect Shots"
+    hosts = ["flame"]
+    families = ["shot"]
+
+    SHARED_KEYS = (
+        "folderPath",
+        "fps",
+        "handleStart",
+        "handleEnd",
+        "item",
+        "otioClip",
+        "resolutionWidth",
+        "resolutionHeight",
+        "retimedHandles",
+        "retimedFramerange",
+        "path",
+        "pixelAspect",
+        "sourceFirstFrame",
+        "versionData",
+        "workfileFrameStart",
+        "xml_overrides",
+    )
+
+    # TODO: add to own plugin for Flame
+    # TODO: toggle for marking task which should be used for product parent
+    add_tasks = []    
+
+    @classmethod
+    def _inject_editorial_shared_data(cls, instance):
+        """
+        Args:
+            instance (obj): The publishing instance.
+        """
+        context = instance.context
+        instance_id = instance.data["instance_id"]
+
+        # Inject folderPath and other creator_attributes to ensure
+        # new shots/hierarchy are properly handled.
+        creator_attributes = instance.data['creator_attributes']
+        instance.data.update(creator_attributes)
+
+        # Inject/Distribute instance shot data as editorialSharedData
+        # to make it available for clip/plate/audio products
+        # in sub-collectors.
+        if not context.data.get("editorialSharedData"):
+            context.data["editorialSharedData"] = {}
+
+        context.data["editorialSharedData"][instance_id] = {
+            key: value for key, value in instance.data.items()
+            if key in cls.SHARED_KEYS
+        }
+
+    def process(self, instance):
+        """
+        Args:
+            instance (pyblish.Instance): The shot instance to update.
+        """
+        instance.data["integrate"] = False  # no representation for shot
+
+        # Adjust instance data from parent otio timeline.
+        otio_timeline = instance.context.data["otioTimeline"]
+        otio_clip, marker = self.get_marker_from_clip_index(
+            otio_timeline, instance.data["clip_index"]
+        )
+        if not otio_clip:
+            raise RuntimeError("Could not retrieve otioClip for shot %r", instance)
+
+        # Compute fps from creator attribute.
+        creator_attrs = instance.data['creator_attributes']
+        if creator_attrs["fps"] == "from_selection":
+            creator_attrs["fps"] = instance.context.data["fps"]
+
+        # Retrieve AyonData marker for associated clip.
+        instance.data["otioClip"] = otio_clip
+
+        # Compute additional data
+        for segment_item in instance.context.data["flameSelectedSegments"]:
+            data = ayfapi.get_segment_data_marker(segment_item) or {}
+            if data.get("clip_index") == instance.data["clip_index"]:
+                break
+
+        else:
+            raise ValueError("Could not retrieve source from selected segments.")
```

Comment:
Is this something that is "artist" error or "code" error? If it is something that artist should know about then use `PublishError` (`from ayon_core.pipeline import PublishError`). It will show the error message to user, otherwise he'll see "This is not your fault".

Reply:
This is adjusted with a `for-if` and the `PublishError`.